framework/core/instance.cpp

```diff
@@ -381,20 +381,54 @@ void Instance::query_gpus()
 
 PhysicalDevice &Instance::get_suitable_gpu()
 {
-	assert(!gpus.empty() && "No physical devices were found on the system.");
+	if (gpus.size() == 0)
+	{
+		throw std::runtime_error("No physical devices found.");
+	}
+
+	uint32_t count_device_type[VK_PHYSICAL_DEVICE_TYPE_CPU + 1];
+	memset(count_device_type, 0, sizeof(count_device_type));
```

Comment:
```[VK_PHYSICAL_DEVICE_TYPE_END_RANGE + 1]``` ?

Reply:
There is no such thing as `VK_PHYSICAL_DEVICE_TYPE_END_RANGE`.

Reply:
This will, as you know, fail if a new device type is introduced in the future. I would instead suggest you use the size of the list you have created below? `device_type_preference` wdyt?